### PR TITLE
Fixes #18741: Add solaris specific paths

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/18741

### CODE REVIEW

File: tree/20_cfe_basics/def.cf [hunk 1]
```diff
@@ -20,4 +20,6 @@ bundle common ncf_def {
   vars:
       "package_module_query_installed_ifelapsed" string => "${node.properties[rudder][packages][installed_cache_expire]}";
       "package_module_query_updates_ifelapsed"   string => "${node.properties[rudder][packages][updates_cache_expire]}";
+  classes:
+      "mpf_stdlib_use_posix_utils" expression => "solaris";
```
Comment (amousset) [inline tree/20_cfe_basics/def.cf:24]: is this class used elsewhere is cfengine stdlib?

Comment (Fdall) [reply tree/20_cfe_basics/def.cf:24]: This is only used in the cfengine paths lib to enable posix tools.

File: tests/acceptance/30_generic_methods/file_content.cf [hunk 2]
```diff
@@ -22,7 +22,13 @@ body common control
 bundle agent init
 {
   vars:
+    solaris::
+      "grep"         string => "/usr/xpg4/bin/grep";
```
Comment (amousset) [inline tests/acceptance/30_generic_methods/file_content.cf:26]: why not using ncf_path for this?

**Comment by Fdall** [comment]
PR updated with a new commit

File: tests/acceptance/30_generic_methods/file_lines_present.cf [hunk 3]
```diff
@@ -41,7 +41,12 @@ bundle agent test
 bundle agent check
 {
   vars:
-    "line_exists_test" string => "/bin/grep -E \"^${init.line_to_add}$\" \"${init.file}\"";
+    solaris::
+      "grep" string => "/usr/xpg4/bin/grep";
```
Comment (amousset) [inline tests/acceptance/30_generic_methods/file_lines_present.cf:45]: same for other tests

**Comment by Fdall** [comment]
PR updated with a new commit

**Review by amousset — APPROVED** [review]

**Pull request merged**